fiat-sql/src/main/kotlin/com/netflix/spinnaker/fiat/permissions/sql/TableDefinitions.kt

```diff
@@ -17,31 +17,145 @@
+    val RESOURCE_TYPE: TableField<PermissionTableRecord, ResourceType> = createField(name("resource_type"), SQLDataType.VARCHAR(255).nullable(false), this, "", ResourceTypeConverter())
+    val RESOURCE_NAME: TableField<PermissionTableRecord, String> = createField(name("resource_name"), SQLDataType.VARCHAR(255).nullable(false), this, "")
+
+    private constructor(alias: Name, aliased: Table<PermissionTableRecord>?): this(alias, null, null, aliased, null)
+
+    constructor(): this(name("fiat_permission"), null)
+
+    override fun `as`(alias: String): PermissionTable = PermissionTable(name(alias), this)
+    override fun `as`(alias: Name): PermissionTable = PermissionTable(alias, this)
+}
+
+class ResourceTableRecord() : TableRecordImpl<ResourceTableRecord>(Tables.RESOURCE) {
+
+}
+
+class ResourceTable(
+    alias: Name,
+    child: Table<out Record>?,
+    path: ForeignKey<out Record, ResourceTableRecord>?,
+    aliased: Table<ResourceTableRecord>?,
+    parameters: Array<Field<*>?>?
+) : TableImpl<ResourceTableRecord>(
+    alias,
+    null,
+    child,
+    path,
+    aliased,
+    parameters,
+    comment(""),
+    TableOptions.table()
+) {
+
+
+    override fun getRecordType(): Class<ResourceTableRecord> = ResourceTableRecord::class.java
+
+    val RESOURCE_TYPE: TableField<ResourceTableRecord, ResourceType> = createField(name("resource_type"), SQLDataType.VARCHAR(255).nullable(false), this, "", ResourceTypeConverter())
+    val RESOURCE_NAME: TableField<ResourceTableRecord, String> = createField(name("resource_name"), SQLDataType.VARCHAR(255).nullable(false), this, "")
+    val BODY: TableField<ResourceTableRecord, String> = createField(name("body"), SQLDataType.LONGVARCHAR.nullable(false), this, "")
+
+    private constructor(alias: Name, aliased: Table<ResourceTableRecord>?): this(alias, null, null, aliased, null)
+
+    constructor(): this(name("fiat_resource"), null)
+
+    override fun `as`(alias: String): ResourceTable = ResourceTable(name(alias), this)
+    override fun `as`(alias: Name): ResourceTable = ResourceTable(alias, this)
+}
+
+class ResourceTypeConverter : AbstractConverter<String, ResourceType>(String::class.java, ResourceType::class.java) {
+
+    override fun from(databaseObject: String?): ResourceType? {
+        return when (databaseObject) {
+            null -> null
+            "${ResourceType.ACCOUNT}" -> ResourceType.ACCOUNT
+            "${ResourceType.APPLICATION}" -> ResourceType.APPLICATION
+            "${ResourceType.BUILD_SERVICE}" -> ResourceType.BUILD_SERVICE
+            "${ResourceType.ROLE}" -> ResourceType.ROLE
+            "${ResourceType.SERVICE_ACCOUNT}" -> ResourceType.SERVICE_ACCOUNT
+            else -> throw IllegalArgumentException("unknown resource type $databaseObject")
+        }
+    }
+
+    override fun to(userObject: ResourceType?): String? {
+        return userObject?.toString()
```

Comment:
Thoughts on moving breaking this out to a new package like `com.netflix.spinnaker.fiat.permissions.sql.tables`?  And then have separate files for each table and table record?  Might make it a bit easier to parse.

Reply:
Should be much nicer now.